src/eventer/eventer_impl.c

```diff
@@ -1260,4 +1304,117 @@ int eventer_thread_check(eventer_t e) {
+    /* fake a aco read */
+    return eventer_aco_read((eventer_aco_t)e, buff, len, NULL);
+  }
+  mtevAssert(aco_get_co() == NULL || aco_get_co() == get_my_impl_data()->aco_main_co);
+  return eventer_fd_opset_get_read(eventer_get_fd_opset(e))
+         (e->fd, buff, len, mask, e);
+}
+
+int eventer_write(eventer_t e, const void *buff, size_t len, int *mask) {
+  if(e->opset == eventer_aco_fd_opset) {
+    /* fake a aco write */
+    return eventer_aco_write((eventer_aco_t)e, buff, len, NULL);
+  }
+  mtevAssert(e->opset != eventer_aco_fd_opset);
+  mtevAssert(aco_get_co() == NULL || aco_get_co() == get_my_impl_data()->aco_main_co);
+  return eventer_fd_opset_get_write(eventer_get_fd_opset(e))
+         (e->fd, buff, len, mask, e);
+}
+
+int eventer_close(eventer_t e, int *mask) {
+  if(e->opset == eventer_aco_fd_opset) {
+    /* fake a aco close */
+    return eventer_aco_close((eventer_aco_t)e);
+  }
+  mtevAssert(aco_get_co() == NULL || aco_get_co() == get_my_impl_data()->aco_main_co);
+  return eventer_fd_opset_get_close(eventer_get_fd_opset(e))
+         (e->fd, mask, e);
+}
+
+int eventer_aco_accept(eventer_aco_t e, struct sockaddr *addr, socklen_t *len, struct timeval *timeout) {
+  mtevAssert(e->opset == eventer_aco_fd_opset);
+  mtevAssert(aco_get_co() != NULL);
+  int mask;
+  struct aco_cb_ctx *ctx = aco_get_arg();
+  if(timeout) ctx->timeout = timeout;
+  return eventer_fd_opset_get_accept(eventer_get_fd_opset((eventer_t)e))
+         (e->fd, addr, len, &mask, (eventer_t)e);
+}
+
+int eventer_aco_read(eventer_aco_t e, void *buff, size_t len, struct timeval *timeout) {
+  mtevAssert(e->opset == eventer_aco_fd_opset);
+  mtevAssert(aco_get_co() != NULL);
+  int mask;
+  struct aco_cb_ctx *ctx = aco_get_arg();
+  if(timeout) ctx->timeout = timeout;
+  return eventer_fd_opset_get_read(eventer_get_fd_opset((eventer_t)e))
+           (e->fd, buff, len, &mask, (eventer_t)e);
+}
+
+int eventer_aco_write(eventer_aco_t e, const void *buff, size_t len, struct timeval *timeout) {
+  mtevAssert(e->opset == eventer_aco_fd_opset);
+  mtevAssert(aco_get_co() != NULL);
+  int mask;
+  struct aco_cb_ctx *ctx = aco_get_arg();
+  if(timeout) ctx->timeout = timeout;
+  return eventer_fd_opset_get_write(eventer_get_fd_opset((eventer_t)e))
+         (e->fd, buff, len, &mask, (eventer_t)e);
+}
+
+int eventer_aco_close(eventer_aco_t e) {
+  mtevAssert(e->opset == eventer_aco_fd_opset);
+  mtevAssert(aco_get_co() != NULL);
+  int mask;
+  return eventer_fd_opset_get_close(eventer_get_fd_opset((eventer_t)e))
+         (e->fd, &mask, (eventer_t)e);
+}
+
+void eventer_aco_free(eventer_aco_t e) {
+  eventer_free((eventer_t)e);
+}
+
+void eventer_aco_start(void (*func)(void), void *closure) {
+  struct eventer_impl_data *t = get_my_impl_data();
+  mtevAssert(t);
+  eventer_aco_setup(t);
+  struct aco_cb_ctx *ctx = calloc(1, sizeof(*ctx));
+  ctx->closure = closure;
+  aco_t *co = aco_create(t->aco_main_co, t->aco_sstk, 32*1024, func, ctx);
```

Comment:
make a define for 32K save stack size and maybe move this to a compile time choice just in case we run into issues there?